custom/crediblex/portfolio/loc/src/main/java/com/crediblex/fineract/portfolio/loc/service/LineOfCreditWritePlatformServiceImpl.java

```diff
@@ -271,26 +271,47 @@ private LineOfCredit convertRequestToEntity(LineOfCreditRequest request, Client
     @Transactional
     public CommandProcessingResult updateLineOfCredit(Long lineOfCreditId, JsonCommand command) {
         try {
+            // Validate limit reduction if maximumAmount is being updated (do this first for business rule validation)
+            if (command.hasParameter("maximumAmount")) {
+                try {
+                    final BigDecimal newMaximumAmount = command.bigDecimalValueOfParameterNamed("maximumAmount");
+                    this.dataValidator.validateForLimitReduction(lineOfCreditId, newMaximumAmount);
+                } catch (Exception e) {
+                    throw new PlatformApiDataValidationException("error.msg.line.of.credit.update.failed",
+                        "Error processing maximumAmount parameter: " + e.getMessage(), "maximumAmount", e);
+                }
+            }
+
             this.dataValidator.validateForUpdate(command.json());
 
             final LineOfCredit lineOfCredit = this.lineOfCreditRepository.findById(lineOfCreditId)
                     .orElseThrow(() -> new PlatformApiDataValidationException("error.msg.line.of.credit.not.found",
                             "Line of credit not found", "lineOfCreditId"));
 
-            final Map<String, Object> changes = lineOfCredit.update(command);
+            final Map<String, Object> changes;
+            try {
+                changes = lineOfCredit.update(command);
+                log.info("Update changes calculated: {}", changes);
+            } catch (Exception updateException) {
+                throw new PlatformApiDataValidationException("error.msg.line.of.credit.update.failed",
+                    "Line of credit update failed during entity update: " + updateException.getMessage(), 
+                    "lineOfCreditId", updateException);
+            }
 
             if (!changes.isEmpty()) {
                 this.lineOfCreditRepository.save(lineOfCredit);
+            } else {
+                log.info("No changes detected for line of credit ID: {}", lineOfCreditId);
             }
 
-
             // Create a map with the updated line of credit data
             final Map<String, Object> responseData = new HashMap<>();
             responseData.put("changes", changes);
```

Comment:
You dont use reponse data anywhere. Remove it. CommandProcessingResultBuilder already gives you a field to use for changes.